tasks/build/download_node_builds.js

```diff
@@ -1,84 +1,126 @@
     });
+  };
 
-    // use an async iife to store promise for download
-    // then store platform in active downloads list
-    // which we will read from in the finish task
-    platform.downloadPromise = (async function () {
-      grunt.file.mkdir(downloadDir);
+  const checkShaSum = (platform) => {
+    const file = basename(platform.nodeUrl);
+    const path = resolve(platform.nodeDir, file);
+    const expected = {
+      hash: 'sha256',
+      expected: platform.win ? shaSums[basename(dirname(platform.nodeUrl)) + '/' + file] : shaSums[file]
+    };
 
-      if (platform.win) {
-        await fromNode(cb => {
-          resp
-          .pipe(createWriteStream(resolve(downloadDir, 'node.exe')))
-          .on('error', cb)
-          .on('finish', cb);
-        });
+    if (!grunt.file.isDir(platform.nodeDir)) {
+      return false;
+    }
+
+    return checkHashFromFileAsync(path, expected).then(([passed, actual]) => {
+      if (passed) {
+        grunt.log.ok(`${platform.name} shasum verified`);
       } else {
-        await fromNode(cb => {
-          resp
-          .pipe(createGunzip())
-          .on('error', cb)
-          .pipe(new Extract({ path: downloadDir, strip: 1 }))
-          .on('error', cb)
-          .on('end', cb);
-        });
+        grunt.log.error(`${platform.name} shasum check failed`);
       }
+      return passed;
+    });
+  };
 
-      await fromNode(cb => {
-        rename(downloadDir, finalDir, cb);
-      });
-    }());
+  const getNodeBuild = (platform) => {
+    const downloadDir = platform.nodeDir;
+    const file = basename(platform.nodeUrl);
+
+    if (grunt.file.isDir(platform.nodeDir)) {
+      grunt.file.delete(platform.nodeDir);
+    }
 
-    activeDownloads.push(platform);
+    grunt.log.ok(`downloading ${platform.name}`);
+
+    return wreckGetAsync(platform.nodeUrl)
+    .then(([resp, payload]) => {
+      if (resp.statusCode !== 200) {
+        throw new Error(`${platform.nodeUrl} failed with a ${resp.statusCode} response`);
+      }
+      grunt.file.mkdir(downloadDir);
+      return payload;
+    })
+    .then(payload => writeFileAsync(resolve(downloadDir, file), payload));
 
-    var bytes = parseInt(resp.headers['content-length'], 10) || 'unknown number of';
-    var mb = ((bytes / 1024) / 1024).toFixed(2);
-    grunt.log.ok(`downloading ${platform.name} - ${mb} mb`);
   };
 
-  grunt.registerTask('_build:downloadNodeBuilds:start', function () {
-    map(platforms, start).nodeify(this.async());
-  });
+  const extractNodeBuild = async (platform) => {
+    const file = basename(platform.nodeUrl);
+    const path = resolve(platform.nodeDir, file);
 
-  grunt.registerTask('_build:downloadNodeBuilds:finish', function () {
-    map(activeDownloads, async (platform) => {
-      await platform.downloadPromise;
-      grunt.log.ok(`${platform.name} download complete`);
-    })
-    .nodeify(this.async());
+    return new Promise((resolve, reject) => {
```

Comment:
The logic for extracting the non-windows node binaries is leaving the original archives in place, so the resulting non-windows kibana archives are about 10-15MB larger than they should be.  You'll need to remove the archive when the extract finishes.

On a related note, I'd recommend using a similar approach to the archives task to execute these actions rather than doing it through a lower level stream: https://github.com/elastic/kibana/blob/master/tasks/build/archives.js#L10-L21

It just seems a little more expressive:

```js
await exec('tar', ['-xzf', path]);
await exec('rm', [path]);
```